### ISSUE TITLE
[MODEL] Implement Random Forest baseline model (Cell 104)

### ISSUE THREAD

**Issue by altivection01** [altivection01/JH_AI_P2#23]
## What's missing
Cell 104 is completely empty under '### Baseline Model' (Random Forest section, Cell 103).

## Required implementation
```python
from sklearn.ensemble import RandomForestClassifier

rf_baseline = RandomForestClassifier(
    n_estimators=100,
    random_state=7,
    n_jobs=-1,
)

pipe_rf_baseline = Pipeline([
    ('preprocess', preprocess),
    ('model', rf_baseline),
])
pipe_rf_baseline.fit(X_train, y_train)

y_val_proba = pipe_rf_baseline.predict_proba(X_val)[:, 1]
y_val_pred  = pipe_rf_baseline.predict(X_val)

print('=== Random Forest — Baseline ===')
evaluate_binary(y_val, y_val_pred, 'RF Baseline')
pr_plot(y_val, y_val_proba, label='RF Baseline')
model_performance_classification(pipe_rf_baseline, X_val, y_val)
```

## Acceptance criteria
- [ ] RF baseline fitted and evaluated on validation set
- [ ] PR-AUC compared to Decision Tree baseline in a written observation
